main.nf

```diff
@@ -222,244 +252,362 @@ if (params.help){
     exit 0
 }
 
-/*
-* SANITY CHECKING
-*/
+// Small console separator to make it easier to read errors after launch
+println ""
 
-// Validate inputs
-if ( params.fasta.isEmpty () ){
-    exit 1, "Please specify --fasta with the path to your reference"
-} else if("${params.fasta}".endsWith(".gz")){
-    //Put the zip into a channel, then unzip it and forward to downstream processes. DONT unzip in all steps, this is inefficient as NXF links the files anyways from work to work dir
-    zipped_fasta = file("${params.fasta}")
+////////////////////////////////////////////////////
+/* --          VALIDATE INPUTS                 -- */
+////////////////////////////////////////////////////
+/**FASTA input handling
+**/ 
 
-    rm_gz = params.fasta - '.gz'
-    lastPath = rm_gz.lastIndexOf(File.separator)
-    bwa_base = rm_gz.substring(lastPath+1)
+if (params.fasta) {
+    file(params.fasta, checkIfExists: true)
+    lastPath = params.fasta.lastIndexOf(File.separator)
+    lastExt = params.fasta.lastIndexOf(".")
+    fasta_base = params.fasta.substring(lastPath+1)
+    index_base = params.fasta.substring(lastPath+1,lastExt)
+    if (params.fasta.endsWith('.gz')) {
+        fasta_base = params.fasta.substring(lastPath+1,lastExt)
+        index_base = fasta_base.substring(0,fasta_base.lastIndexOf("."))
 
+    }
+} else {
+    exit 1, "[nf-core/eager] error: please specify --fasta with the path to your reference"
+}
+
+// Validate reference inputs
+if("${params.fasta}".endsWith(".gz")){
     process unzip_reference{
         tag "${zipped_fasta}"
 
         input:
-        file zipped_fasta
+        path zipped_fasta from file(params.fasta) // path doesn't like it if string of object not prefaced with root dir (/), so use file to find 
```

Comment:
```suggestion
        path zipped_fasta from file(params.fasta) // path doesn't like it if a string of an object is not prefaced with a root dir (/), so use file to find 
```
This is a bit confusing to me, because you switched from `file` to `find` but the comment suggest it the other way around

Reply:
Actually it's referring have to do the `from file(params.fasta) ` file before `path` (this is my personal note)